local-cli/runWindows/utils/msbuildtools.js

```diff
@@ -57,19 +64,38 @@ class MSBuildTools {
 
 function checkMSBuildVersion(version) {
   const query = `reg query HKLM\\SOFTWARE\\Microsoft\\MSBuild\\ToolsVersions\\${version} /s /v MSBuildToolsPath`;
+  let toolsPath = null;
+  // Try to get the MSBuild path using registry
   try {
     const output = child_process.execSync(query).toString();
-    let toolsPath = /MSBuildToolsPath\s+REG_SZ\s+(.*)/i.exec(output);
-    if (toolsPath) {
-      toolsPath = toolsPath[1];
+    let toolsPathOutput = /MSBuildToolsPath\s+REG_SZ\s+(.*)/i.exec(output);
+    if (toolsPathOutput) {
+      toolsPathOutput = toolsPathOutput[1];
       // Win10 on .NET Native uses x86 arch compiler, if using x64 Node, use x86 tools
       if ((version === '15.0' || version === '14.0' && toolsPath.indexOf('amd64') > -1)) {
-        toolsPath = path.resolve(toolsPath, '..');
+        toolsPathOutput = path.resolve(toolsPathOutput, '..');
       }
-      console.log(chalk.green(`Found MSBuild v${version} at ${toolsPath}`));
-      return new MSBuildTools(version, toolsPath);
+      toolsPath = toolsPathOutput;
     }
   } catch (e) {
+    toolsPath = null;
+  }
+  // Try to get the MSBuild Path by checking file system (for VS2017)
+  if (!toolsPath) {
```

Comment:
VS2017 removed registry variables so you can't locate VS the same way anymore. See: https://docs.microsoft.com/en-us/visualstudio/msbuild/what-s-new-in-msbuild-15-0

Also, I found this today: https://github.com/Microsoft/vswhere so I am going to switch to using that exe once I can package it up as a separate npm module that we can depend on.